extensions/AzureCosmosDBMongoVCore/AzureCosmosDBMongoVCore/AzureCosmosDBMongoVCoreMemory.cs

```diff
@@ -0,0 +1,299 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Configuration;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.KernelMemory.AI;
+using Microsoft.KernelMemory.Diagnostics;
+using Microsoft.KernelMemory.MemoryStorage;
+using MongoDB.Driver;
+using MongoDB.Bson;
+
+namespace Microsoft.KernelMemory.MemoryDb.AzureCosmosDBMongoVCore;
+
+/// <summary>
+/// Azure CosmosDB Mongo vCore connector for Kernel Memory
+/// </summary>
+
+public class AzureCosmosDBMongoVCoreMemory : IMemoryDb
+{
+    private readonly ITextEmbeddingGenerator _embeddingGenerator;
+    private readonly ILogger<AzureCosmosDBMongoVCoreMemory> _log;
+    private readonly AzureCosmosDBMongoVCoreConfig _config;
+    private readonly IMongoClient _cosmosDBMongoClient;
+    private readonly IMongoDatabase _cosmosMongoDatabase;
+    private readonly IMongoCollection<AzureCosmosDBMongoVCoreMemoryRecord> _cosmosMongoCollection;
+    private readonly String _collectionName;
+
+    /// <summary>
+    /// Create a new instance
+    /// </summary>
+    /// <param name="config"> Azure Cosmos DB Mongo vCore configuration</parama>
+    /// <param name="embeddingGenerator">Text embedding generator</param>
+    /// <param name="log">Application logger</param>
+    public AzureCosmosDBMongoVCoreMemory(
+        AzureCosmosDBMongoVCoreConfig config,
+        ITextEmbeddingGenerator embeddingGenerator,
+        String databaseName,
+        String collectionName,
+        ILogger<AzureCosmosDBMongoVCoreMemory>? log = null) 
+    {
+            this._embeddingGenerator = embeddingGenerator;
+            this._log = log ?? DefaultLogger<AzureCosmosDBMongoVCoreMemory>.Instance;
+            this._config = config;
+
+            if (string.IsNullOrEmpty(config.ConnectionString)) {
+                this._log.LogCritical("Azure Cosmos DB Mongo vCore connection string is empty.");
+                throw new AzureCosmosDBMongoVCoreMemoryException("Azure Cosmos DB Mongo vCore connection string is empty.");
+            }
+
+            if (this._embeddingGenerator == null) {
+                throw new AzureCosmosDBMongoVCoreMemoryException("Embedding generator not configured");
+            }
+
+            this._cosmosDBMongoClient = new MongoClient(config.ConnectionString);
```

Comment:
Create MongoClientSettings first and then the MongoClient so you can pass the appName

e.g.
MongoClientSettings settings = MongoClientSettings.FromConnectionString(config.ConnectionString);
settings.ApplicationName = config.ApplicationName;
this._cosmosDBMongoClient = new MongoClient(settings);

Reply:
Added the code creating AppName.